app/controllers/actions/RequiredUserAnswersForCorrectReturn.scala

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.actions
+
+import models.correctReturn.{AddASmallProducer, RepaymentMethod}
+import models.requests.DataRequest
+import models.{CheckMode, LitresInBands, NormalMode, RetrievedSubscription, SdilReturn}
+import pages.{Page, QuestionPage}
+import pages.correctReturn._
+import play.api.libs.json.Reads
+import play.api.mvc.Result
+import play.api.mvc.Results.Redirect
+import utilities.{GenericLogger, UserTypeCheck}
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+import scala.reflect.ClassTag
+
+class RequiredUserAnswersForCorrectReturn @Inject()(genericLogger: GenericLogger)(implicit val executionContext: ExecutionContext) extends ActionHelpers {
+
+  def requireData(page: Page)(action: => Future[Result])(implicit request: DataRequest[_]): Future[Result] = {
+    page match {
+      case CorrectReturnBaseCYAPage => checkYourAnswersRequiredData(action)
+      case CorrectReturnCheckChangesPage => checkChangesRequiredData(action)
+      case _ => action
+    }
+  }
+
+  private[controllers] def checkYourAnswersRequiredData(action: => Future[Result])(implicit request: DataRequest[_]): Future[Result] = {
+    val userAnswersMissing: List[CorrectReturnRequiredPage[_, _, _]] = returnMissingAnswers(mainRoute)
```

Comment:
Why are we using a CorrectReturnPreviousPage rather than a more generic PreviousPage? (Seems out of step with ChangeActivity not requiring one)

Reply:
As discussed, not needed.